### ISSUE TITLE
Export to a folder instead of a file

### ISSUE THREAD

**Issue by jornfranke** [scrapy/scrapy#5556]
## Summary

Currently, feedexports can only be to one specific file (see [here](https://docs.scrapy.org/en/latest/topics/feed-exports.html#storage-uri-parameters). While this fine for many cases and should be kept, there should be also an option to store only in a specific folder and all subfolders/files of this specific folder should be considered the result of a feedexport. For example, I have written a feed exporter (see [Codeberg](https://codeberg.org/ZuInnoTe/scrapy-contrib-bigexporters) or [Github](https://github.com/ZuInnoTe/scrapy-contrib-bigexporters)) that allows exporting to parquet, avro or orc to one file. This works excellent even for very large files.

However, other formats require to store metadata and data as part of (potential many) different files in subfolders. For instance, Apache Iceberg: https://github.com/ZuInnoTe/scrapy-contrib-bigexporters/issues/7 (see also its [spec](https://iceberg.apache.org/spec/)), which is a novel Big Data format that is also optimized for object storages, such as S3. 

Hence, it would be nice to allow to specify an output folder instead of a file and any follow up processes should consider the output folder including all subdirectories/files. Maybe one should also consider that the content of the output directory is only complete if the feedexporter creates an empty SUCCESS file in this output folder or similar.

## Motivation

Support of novel output formats to write avoiding unnecessary transformations afterwards and thus reducing the implementation effort later-on.

## Describe alternatives you've considered

One could only specify the output directory and not a file as part of the URI and then in the exporter check if a file or folder is specified, but this may not be so trivial. Furthermore, it is break of the API, which according to the documentation and all examples clearly expects a file. Finally, other tools building on top of Scrapy may not be able to deal with the fact that the output is a folder and one needs to consider all subfolders/files.

One could also zip/tar.gz a folder containing subfolders and files, but then it has to be later on again unzipped to process further.

## Additional context

Any additional information about the feature request here.

**Comment by yogesh-9999** [comment]
!assign

**Comment by pioneerHitesh** [comment]
I would like to contribute , can you please assign it to me?

**Comment by Gallaecio** [comment]
I don’t think this is a good choice for a first time contribution to Scrapy, and the right solution might require some API discussion before an implementation is proposed.

**Comment by kangqiwang** [comment]
is there any updates for iceberg, are we going to support this format ?

**Comment by wRAR** [comment]
@kangqiwang you may be misunderstanding something

**Comment by wRAR** [comment]
@kangqiwang if you are asking about support of Iceberg in Scrapy itself then that wasn't planned, if you are asking about e.g. `scrapy-contrib-bigexporters` the correct place to ask questions about it in its repo, not in the Scrapy one, and if you are asking about plans for implementing folder export in Scrapy, which is what this issue is about, then there are no updates or plans to do this, so somebody interested in it needs to propose code or at least ideas for it.

**Comment by jornfranke** [comment]
I work currently on integrating iceberg into scrapy-contrib-bigexporter.

Meanwhile, Iceberg has evolved and it supports many different technologies (not only filesystems). Thus, I do not think the feature I requested here make sense anymore.

Nevertheless, I plan to integrate Iceberg into the coming weeks in scrapy-contrib-bigexporter.
However, I will not be able to write to the file provided by Scrapy, but will write to the catalog and tables configured in pyiceberg (you can compare the scenario to writing to a database from Scrapy).
